test/tests/functional/pbs_reservations.py

```diff
@@ -1123,3 +1124,407 @@ def test_reservations_with_expired_subjobs(self):
+        self.server.expect(JOB, {'job_state': 'B'}, jids[0])
+        self.server.expect(JOB, {'job_state=R': 2}, count=True,
+                           id=jids[0], extend='t')
+        self.server.expect(JOB, {'job_state=R': 2}, count=True,
+                           id=jids[1], extend='t')
+        self.server.expect(JOB, {'job_state=Q': 3}, count=True,
+                           id=jids[2], extend='t')
+        # At end of first instance of reservation ,verify running sub-job
+        # should be finished
+        self.logger.info(
+            'Waiting 20 sec job-array 1 and 2 to be finished')
+        self.server.expect(JOB, {'job_state=F': 3}, extend='xt',
+                           offset=20, id=jids[0])
+        self.server.expect(JOB, {'job_state=F': 3}, extend='xt',
+                           id=jids[1])
+
+        # Wait for standing reservation second instance to confirmed
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, exp_attr, id=rid)
+        # Check for queued jobs in second instance of reservation
+        self.server.expect(JOB, {'job_state': 'Q',
+                                 'comment': (MATCH_RE, 'Queue not started')},
+                           id=jids[2])
+        self.logger.info(
+            'Waiting 55 sec for second instance of reservation to start')
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, exp_attr, id=rid, offset=55, interval=1)
+        self.server.expect(RESV, {'reserve_index': 2}, id=rid)
+        # Check for queued jobs should be running
+        self.server.expect(JOB, {'job_state=R': 2}, extend='xt',
+                           id=jids[2])
+
+        # Check for running jobs in second instance should finished
+        self.logger.info(
+            'Waiting 30 sec for second instance of reservation to finished')
+        self.server.expect(JOB, {'job_state=F': 3}, extend='xt',
+                           offset=30, id=jids[2])
+
+        # Wait for reservation to be finished
+        msg = "Que;" + rid_q + ";deleted at request of pbs_server@"
+        self.server.log_match(msg, starttime=end, interval=2)
+        for job in jids:
+            self.server.expect(JOB, 'queue', op=UNSET, id=job)
+
+        # At end of reservation,verify running sub-jobs from job-array 3
+        # terminated
+        self.server.expect(JOB, {'job_state': 'F', 'substate': '91'},
+                           extend='x', id=jids[2])
+        self.server.expect(JOB, {'job_state': 'F', 'substate': '91',
+                                 'queue': rid_q}, extend='x',
+                           attrop=PTL_AND, id=subjid[5])
+
+        # Check for  sub-jobs status of job-array 1 and 2
+        # as all sub-jobs from job-array 1 and 2 exceed walltime
+        # and failed to complete within an instance of reservation
+        # so it should substate as 93
```

Comment:
exceed walltime of the reservation, so they will not complete running, so the substate of these subjobs should be 93.